proto/src/determined/trial/v1/trial.proto

```diff
@@ -30,6 +30,30 @@ message CheckpointWorkload {
   google.protobuf.Struct metadata = 9;
 }
 
+// The current state of the trial. see \dT+ trial_state in db
+enum State {
```

Comment:
```
 determined> \dT+ trial_state
 
 PAUSED             
 STOPPING_CANCELED  
 STOPPING_KILLED    
 STOPPING_COMPLETED 
 STOPPING_ERROR     
 CANCELED           
 COMPLETED          
 ERROR              


determined> \dT+ experiment_state

PAUSED             
STOPPING_CANCELED  
STOPPING_KILLED    
STOPPING_COMPLETED 
STOPPING_ERROR     
DELETING           
CANCELED           
COMPLETED          
ERROR              
DELETE_FAILED      
```

so change `Trial`? 

related: can we override/remove the rule about prefixing with enum with `STATE_`? seems to just make things more error-prone.

Reply:
>so change `Trial`?

that seems right a good way to go, can you give it a try? I am not sure if that'd need some dramatic changes. On JSON API level it shouldn't affect anything.

>related: can we override/remove the rule about prefixing with enum with STATE_? seems to just make things more error-prone.

it is controversial, but 
- we do it everywhere so that be awkward for consistency
- if we change it here it'd be a breaking change for a bunch of Trial APIs.

Reply:
can this be a follow up?

Reply:
(this would require frontend changes as well)

Reply:
Fast follow up is ok